File: john-daniel-jimenez-calculator-2012-05-07-105900/lib/calculator.rb
class Calculator
  def initialize expression=''
    @expression = expression
  end

  def expr
    @expression
  end

  def expr= expression
    @expression = expression
  end

  def add 
    @expression.split(',').inject(0) {|s,v| s += v.to_i}
  end
end

File: daniel-finster-2015-08-24/calculator.py
class Calculator (object):
    expression = None
    digits = []

    def __init__ (self, string):
        self.assign(string)

    def assign (self, string):
        self.expression = string
        if string != "":
            parts = string.split(",")
            self.digits = map(int, parts)

    def expr (self):
        return self.expression

    def add (self):
        return reduce(lambda a,b: a + b, self.digits, 0)
